Lib/test/test_structseq.py

```diff
@@ -93,6 +94,53 @@ def test_constructor(self):
         s = "123456789"
         self.assertEqual("".join(t(s)), s)
 
+    def test_constructor_with_duplicate_fields(self):
+        t = time.struct_time
+        self.assertEqual(t("123456789"), tuple("123456789"))
+        self.assertEqual(t("1234567890"), tuple("123456789"))
+        self.assertEqual(t("1234567890").tm_zone, "0")
+        self.assertEqual(t("123456789", dict={"tm_zone": "some zone"}), tuple("123456789"))
+        self.assertEqual(t("123456789", dict={"tm_zone": "some zone"}).tm_zone, "some zone")
```

Comment:
These tests do not actually test duplicate fields, so it is better to move them to `test_constructor`.

There is a test for positional first argument and keyword "dict" argument. I think that it is better to pass both arguments as positional for most tests, but add also a case for both passed by keyword.